packages/gatsby/cache-dir/loader.js

```diff
@@ -280,7 +273,7 @@ const queue = {
   // if necessary and then the code/data bundles. Used for prefetching
   // and getting resources for page changes.
   getResourcesForPathname: path =>
-    new Promise((resolve, reject) => {
+    new Promise((resolve, reject) => { /* eslint-disable-line consistent-return */
```

Comment:
let's make returns consistent here if possible

Reply:
@pieh But what kind of value should I return in Promise executor callback?
Of course I know executor handle `resolve` or 'reject' depending on the logic but I have no idea properly value of `return` statement.

I referenced MDN example but I can't find it.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

Reply:
Something like this should fix this https://gist.github.com/pieh/dd4e38b87edeba2b642ff994cb83d480 - we don't need to return resolve/reject - that's just shortcut to end execution early, but we can split that in 2

Reply:
@piehi I already tried this solution, but ESLint doesn't allow single `return` statement.
It can solve easily to give value whatever like `return 0`.

<img width="802" alt="screen shot 2018-10-11 at 20 25 45" src="https://user-images.githubusercontent.com/5501268/46800860-f17a2280-cd93-11e8-9196-0edf6a142e00.png">

<img width="1258" alt="screen shot 2018-10-11 at 20 25 14" src="https://user-images.githubusercontent.com/5501268/46800907-0ce52d80-cd94-11e8-87e1-3ca97f5c69a9.png">

How could I solve this? It up to you 😀(or repo coding style)

Reply:
Did you change all the returns? I did this quick change and at least eslint inside vscode stopped reporting this issue

Reply:
> Did you change all the returns?
No 😶

> I  did this quick change and at least eslint inside vscode stopped reporting this issue
hmm... it's strange.

Anyway I've just read ESLInt `consistent-return` Rule [Doc](https://eslint.org/docs/rules/consistent-return) and decide better way.
How about you? 😀

Reply:
Yeah, we don't really care about returning anything from that promise - return there are made only to terminate function early, I would really prefer if we didn't return undefined (that's potential source of WTF when reading that code :) )

Reply:
@pieh Totally :)
I've finished a work in PR.
It's up to you whether close or merge.
Also that brought meaningful meaningful knowledge to me about return value of `Promise` executor reached end line of function without `resolve()`,`reject()` .

Thank you so much😄